File: backend/ecommerce/src/main/java/com/bookstore/ecommerce/app/domain/data/Book.java
package com.bookstore.ecommerce.app.domain.data;

import java.time.Instant;
import lombok.Builder;
import lombok.Getter;

@Builder
public class Book {
  @Getter
  private String id;
  @Getter
  private String title;
  @Getter
  private String subtitle;
  @Getter
  private String description;
  @Getter
  private String[] authors;
  @Getter
  private String publisher;
  @Getter
  private Instant publishedDate;
  @Getter
  private double averageRating;
  @Getter
  private int ratingsCount;
  @Getter
  private String thumbnailUrl;
  @Getter
  private String previewUrl;
  @Getter
  private int onhandQty;
  @Getter
  private int reservedQty;
}

File: store/app/book/query/book.go
package query

import (
	"store/app/domain"
	"time"
)

type Book struct {
	Id            string
	Title         string
	Subtitle      string
	Description   string
	Authors       []string
	Publisher     string
	PublishedDate time.Time
	AverageRating float32
	RatingsCount  int
	ThumbnailUrl  string
	PreviewUrl    string
}

func (Book) fromDataObject(book domain.BookData) Book {
	return Book{
		Id:            book.Id,
		Title:         book.Title,
		Subtitle:      book.Subtitle,
		Description:   book.Description,
		Authors:       book.Authors,
		Publisher:     book.Publisher,
		PublishedDate: book.PublishedDate,
		AverageRating: book.AverageRating,
		RatingsCount:  book.RatingsCount,
		ThumbnailUrl:  book.ThumbnailUrl,
		PreviewUrl:    book.PreviewUrl,
	}
}
